python/pyarrow/_s3fs.pyx

```diff
@@ -149,6 +157,8 @@ cdef class S3FileSystem(FileSystem):
             )
         elif anonymous:
             options = CS3Options.Anonymous()
+        elif use_web_identity:
+            options = CS3Options.FromAssumeRoleWithWebIdentity()
```

Comment:
If `use_web_identity`, `anonymous`, `role_arn` and (`access_key` or `secret_key`) are all exclusive, we should probably test for such cases and raise an error.